src/RON/components/SentPacketLogger.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react';
+import CSS from 'csstype';
+import ReactTable from 'react-table-v6';
+import fs from 'fs';
+// import "../../node_modules/react-table-v6/react-table.css"
+import { rovecomm, RovecommManifest } from '../../Core/RoveProtocol/Rovecomm';
+
+const h1Style: CSS.Properties = {
+  fontFamily: 'arial',
+  fontSize: '18px',
+  margin: '5px 0px',
+};
+
+const container: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'column',
+  fontFamily: 'arial',
+  borderTopWidth: '28px',
+  borderColor: '#990000',
+  borderBottomWidth: '2px',
+  borderStyle: 'solid',
+  padding: '5px',
+  alignItems: 'center',
+  overflow: 'auto',
+};
+const label: CSS.Properties = {
+  marginTop: '-10px',
+  position: 'relative',
+  top: '24px',
+  left: '3px',
+  fontFamily: 'arial',
+  fontSize: '16px',
+  zIndex: 1,
+  color: 'white',
+};
+const selectbox: CSS.Properties = {
+  display: 'flex',
+  flexDirection: 'row',
+  width: '75%',
+  margin: '2.5px',
+  justifyContent: 'space-around',
+};
+const selector: CSS.Properties = {
+  width: '200px',
+};
+
+const buttons: CSS.Properties = {
+  fontFamily: 'arial',
+  lineHeight: '20px',
+  fontSize: '16px',
+};
+
+interface IProps {
+  style?: CSS.Properties;
+}
+
+interface IState {
+  data: any;
+  columns: any;
+}
+
+class SentPacketLogger extends Component<IProps, IState> {
+  static defaultProps = {
+    style: {},
+  };
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      data: [],
+      columns: [
+        { Header: 'Name', accessor: 'name', width: '100' },
+        { Header: 'Data Id', accessor: 'dataId', width: '75' },
+        { Header: 'Time', accessor: 'time', width: '100' },
+        { Header: 'Type', accessor: 'dataType', width: '50' },
+        { Header: 'Count', accessor: 'dataCount', width: '50' },
+        {
+          Header: 'Data',
+          accessor: 'data',
+          width: 'fill',
+          Cell: (data: any) => (
+            <div
+              style={{
+                overflow: 'none',
+                textOverflow: 'ellipsis',
+              }}
+            >
+              {data.value.join(', ')}
+            </div>
+          ),
+        },
+      ],
+    };
+    this.addData = this.addData.bind(this);
+    rovecomm.on('BasestationCommand', (data: any) => this.addData(data));
+  }
+
+  addData(newData: any): void {
+    this.setState((prevState) => ({ data: [newData].concat(prevState.data) }));
+  }
```

Comment:
Should add a clear button or only save up to like 100 packets for this since we don't want it to eat up the entire system's memory over the entire time of a task saving every packet that was sent.